docs/aws/services/Cognito.md

```diff
@@ -0,0 +1,127 @@
+```
+
+Developers may miss the fact that these attributes are controllable by users directly:
+
+- If an account has an admin field that is writable by the user, a developer may not realise this and use that attribute to authorise an account to perform privileged actions. This would lead to a privilege escalation vector.
+- Cognito allows for email addresses to be changed through the CLI. If this is misconfigured to reflect changes before they are confirmed via an email/SMS code, a downstream service could potentially use the unconfirmed address when authenticating the user. This would result in an account takeover vector. See [Flickr account takeover](https://security.lauritz-holtmann.de/advisories/flickr-account-takeover/) for more detail.
+
+**Firstly, when configuring the user pool ensure Keep original attribute value active when an update is pending is enabled. This would stop changes in the email address from being reflected until it is verified. Secondly, read/ write access control can be applied on user attributes. Limit read/write access to attributes which are used in downstream services, especially for authentication and authorisation. Lastly, avoid using the 'email' attribute when authenticating a user. Instead, use the 'sub' attribute, which is guaranteed to be unique for each user.**
+
+### Identity Pools
+
+#### Getting an identity pool ID
+
+When attacking a Cognito identity pool, its identity pool id must first be enumerated. They may be hardcoded in various JavaScript files hosted on the website, and are of the form REGION:GUID.
+
+#### Assuming roles using an Identity pool ID
+
+An unauthenticated role can be assumed through the following:
+
+```bash
+aws cognito-identity get-id --identity-pool-id <identity pool id>
+```
+
+```bash
+aws cognito-identity get-credentials-for-identity --identity-id <identity id>
+```
+
+The credentials from the last command can be added to environment variables, or a profile can be created with it. This would allow for a role to be assumed within the AWS account.
+
+Before assuming a role for an authenticate user, the ID token for an authenticated user must be obtained. This can be done using the methods explained previously. Using an ID token, the following CLI commands can be used to obtain an authenticated role:
+
+```bash
+aws cognito-identity get-id --identity-pool-id <identity pool id> --login <ISS>=<Identity token>
+```
+
+```bash
+cognito-identity get-credentials-for-identity --identity-id <identity id> --login <ISS>=<Identity token>
+```
+
+*\<ISS\> is equivalent to the issuer of the token, which can be found inside the payload of the ID token JWT*
+
+**When configuring an identity pool, apply the principle of least privilege. Disable the unauthenticated role if possible, and apply the minimum privileges required for the authenticated role using IAM policies.**
```

Comment:
Any example policies or AWS docs we can ref here? Or maybe an example IAM policy for a simple webapp that is authing via Cognito?

Reply:
Thing is its so context dependent, I'm not really sure what to give as an example. At this point, its just saying to configure IAM roles properly, which is its whole section on its own. Perhaps I should link back to the IAM page on the wiki or something?